src/common/camera.h

```diff
@@ -16,11 +16,14 @@
 
 class camera {
   public:
-    void initialize(double aspect_ratio = 1.0) {
+    void initialize() {
         auto theta = degrees_to_radians(vfov);
         auto h = tan(theta/2);
         auto viewport_height = 2.0 * h;
-        auto viewport_width = aspect_ratio * viewport_height;
+
+        // Calculate the actual aspect ratio of the integer image width and height.
+        auto actual_aspect_ratio = static_cast<double>(image_width) / get_image_height();
```

Comment:
Not sure how I feel about this.

You're correct that the image_height could be rounded in such a way that the "actual" aspect ratio is different from the parameter

But this looks really messy.

I think it's fine because this is only called once, and it's to solve a possible technical round-off error

I would like this to be neatly documented to the associated text (which I assume you already plan to write).

Do you think that the roundoff error possibility should be added into a comment?